src/data/parsing.py

```diff
@@ -0,0 +1,309 @@
+
+
+    if writeFile:
+        with open(pathToFolders+'/dep/{0:04d}.dep'.format(docNum), "w", encoding='utf8') as text_file:
+            text_file.write('\n\n'.join(l_depSentences))
+        with open(pathToFolders+'/input/{0:04d}.input'.format(docNum), "w", encoding='utf8') as text_file:
+            text_file.write('\n\n'.join(l_posSentences))
+        with open(pathToFolders+'/morph/{0:04d}.morph'.format(docNum), "w", encoding='utf8') as text_file:
+            text_file.write('\n\n'.join(l_morSentences))
+                
+        print('Files written to folder:', pathToFolders)
+    return d_documentData
+
+
+def get_adjustment_position(tokenPosition, adjustmentDictionary):
+    '''This determines the adjustment position for DEP files, because things get reordered when there are equations
+    '''
+    if len(adjustmentDictionary)>1:
+        for key, val in sorted(list(adjustmentDictionary.items()), key=lambda x:x, reverse=True):
+            if tokenPosition>key:
+                return val
+    return 0
+
+
+def get_token_governor(token, sent):
+    if token.governor==0:
+        govWord = 'ROOT'
+    else:
+        govWord = sent.words[token.governor-1].text
+    return govWord
+        
+
+def load_data(jsonPath, picklePath = None):
+    """Load data - if picklePath is specified, load the pickle. Else, try json file.
+    This returns the JSON file as well as a list of document texts 
+    """
+    if picklePath is not None:
+        l_docs = pickle.load(open(picklePath, "rb" ))
+    else:
+
+        ## Read JSON data into the datastore variable - this comes from Peter and Domonique's effort. Don
+        with open(jsonPath, 'r') as f:
+            datastore = json.load(f)
+                  
+        ## These were some bad files - nothing substantive in them, or they were retrieved in bad format
+        for e in ['1805.10677v1', '0911.5378v1']: 
+            datastore.pop(e)
+                
+        ## Extract texts
+        l_docs = [value['text'] for key,value in list(datastore.items())[0:] if value['text'] ]
+        
+    print('# of documents: ', len(l_docs))
+    
+    return datastore, l_docs
+
+
+def run(args_dict):
+    ''' Main run file that orchestrates everything  
+    '''
+    
+    ## Load NLPs
+    spacynlp = spacy.load('en_core_web_sm')
+    nlp = stanfordnlp.Pipeline(models_dir=args_dict['stanfordnlp'], 
+                            treebank='en_ewt', use_gpu=False, pos_batch_size=3000)
+    
+    
+    ## Load documents 
+    jsonObj, allDocs = load_data(args_dict['data'])
+
+    
+    ## Process and Clean documents 
+    try: 
+        allDocsClean = pickle.load(open('allDocsClean.pkl', "rb" ))
+        print('Loaded pickle!')
+    except:
+        print('Starting from scratch')
+        allDocsClean= []
+        for i, doc in enumerate(allDocs):
+            if i%10==0:
+                print(i)
+            allDocsClean.append(textparsing.clean_doc(doc, spacynlp))

```

Comment:
Note, if you did import `clean_doc` as I suggested above, this would need to change to just be `allDocsClean.append(clean_doct(doc, spacynlp))`. I don't think this is a deal breaker though... so not going to hold up the merge.